driver-core/src/main/com/mongodb/internal/operation/CommandOperationHelper.java

```diff
@@ -159,609 +139,331 @@ public T apply(final T t, final AsyncConnectionSource source, final AsyncConnect
         };
     }
 
-    static Function<BsonDocument, BsonDocument> noOpRetryCommandModifier() {
-        return command -> command;
-    }
-
     interface CommandCreator {
         BsonDocument create(ServerDescription serverDescription, ConnectionDescription connectionDescription);
     }
 
-    /* Read Binding Helpers */
-
-    static BsonDocument executeCommand(final ReadBinding binding, final String database, final CommandCreator commandCreator,
-                                       final boolean retryReads) {
-        return executeCommand(binding, database, commandCreator, new BsonDocumentCodec(), retryReads);
+    private static Throwable chooseAndMutateRetryableReadException(
```

Comment:
I wanted to have the "mutate" part in the method name to make it clear to a future maintainer that the method is allowed to mutate its arguments. Otherwise, the only way to learn that this is allowed is to follow the method usage, which will lead to the constructor of `RetryingAsyncCallbackSupplier`, where this is described. The latter is completely fine, but the former seems more helpful and harmless.

Reply:
Renamed :)